10up/ads-txt
# Ensure administrator role exists before modifying.

### Description of the Change


Adds a check that the administrator role exists before attempting to call the `::has_cap()`, `::add_cap()` and `::remove_cap()` methods.

If the administrator role has been deleted from the site, `get_role()` returns `null` and triggers a fatal error.

Without an `administrator` role, no users will be given the capability to edit the ads.txt files until the capability is added to the role.


Closes #139 

### How to test the Change


1. Deactivate the plugin (this will remove the capability)
2. Create a clone of the administrator role via wp-cli: `wp role create admin Admin --clone=administrator`
3. Remove the administrator role via wp-cli: `wp role delete  administrator`
4. Give your user the new role via wp-cli: `wp user set-role [user_id] admin`
5. Activate the plugin

On this branch, the final step should proceed without problems. On `develop` it will throw an error due to the role not existing.

Following the test steps above, your `admin` user will not be able to edit ads.txt as no capability is added.

### Changelog Entry

> Fixed - Fatal error if the role `administrator` does not exist.


### Credits

Props @peterwilsoncc, @renzit, @Sidsector9.


### Checklist:
